doc/manual/src/architecture/store/store.md

```diff
@@ -0,0 +1,212 @@
+# Store
+
+A Nix store is a collection of *store objects* with references between them.
+It supports operations to manipulate that collection.
+
+The following concept map is a graphical outline of this chapter.
+Arrows indicate suggested reading order.
+
+```
+                      ,--------------[ store ]----------------,
+                      |                  |                    |
+                      v                  v                    v
+               [ store object ]     [ closure ]--,      [ operations ]
+                      |               |   |      |        |        |
+                      v               |   |      v        v        |
+           [ files and processes ]    |   | [ garbage collection ] |
+               /          \           |   |                        |
+              v            v          |   v                        v
+[ file system object ] [ store path ] | [ derivation ]--->[ building ]
+                  |        ^      |   |                         |
+                  v        |      v   v                         |
+             [ digest ]----' [ reference scanning ]<------------'
+              /      \
+             v        v
+[ input addressing ] [ content addressing ]
+```
+
+## Store Object {#store-object}
+
+A store object can hold
+
+- arbitrary *data*
+- *references* to other store objects.
+
+A Nix store makes no distinction if store objects are build inputs, build results, or build tasks.
```

Comment:
Yet a ~store object~ store can not build a store path that isn't a derivation.

If we're going to keep the new abstraction, may I suggest "store node"?

Reply:
That sentence is probably confusing. It was meant to say that the store keeps all of these in a collection without distinction. What about that?

```suggestion
Store objects can be build inputs, build results, or build tasks.
```